### PR TITLE
Workflow 376

### PR DESCRIPTION
**Pull request by ErnestaP**

### CODE REVIEW

File: .github/workflows/pytest.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      - uses: actions/checkout@v2
+      - name: Install Python 3
+        uses: actions/setup-python@v1
+        with:
+          python-version: 3.8.10
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r requirements.txt &&
+          pip install "apache-airflow==${AIRFLOW_VERSION}" --constraint "${CONSTRAINT_URL}"
```
Comment (drjova) [inline .github/workflows/pytest.yml]: why these requirements are not in the `requirements.txt` file?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,14 @@
+FROM apache/airflow:latest-python3.7
+
+USER root
```
Comment (drjova) [inline Dockerfile]: do we have to run it in `root` 🤔

File: requirements.txt [hunk 3]
```diff
@@ -0,0 +1 @@
+pytest==7.1.0
```
Comment (drjova) [inline requirements.txt]: I would put this an extra

File: README.md [hunk 4]
```diff
@@ -1 +1,44 @@
...
+```
+    pyenv global 3.7.10
+    export PYTHON_VERSION=3.7.10
+    pyenv install $PYTHON_VERSION
+    pyenv virtualenv $PYTHON_VERSION workflows
+    pyenv activate workflows
+```
+2.  Once you have activated your Python environment, install your repository as a Python package.
+```
+    poetry install
```
Comment (drjova) [inline README.md]: we don't use poetry anymore :)

File: tests/test_dag/__init__.py [hunk 5]
```diff
@@ -0,0 +1 @@
+import dags.test_dag
```
Comment (drjova) [inline tests/test_dag/__init__.py]: You have not endline in the end of the files, that's in all files.

File: README.md [hunk 6]
```diff
@@ -1 +1,44 @@
...
+4. Install Airflow using the constraints file:
+```
+AIRFLOW_VERSION=2.2.4
+PYTHON_VERSION="$(python --version | cut -d " " -f 2 | cut -d "." -f 1-2)"
+CONSTRAINT_URL="https://raw.githubusercontent.com/apache/airflow/constraints-${AIRFLOW_VERSION}/constraints-${PYTHON_VERSION}.txt"
+pip install "apache-airflow==${AIRFLOW_VERSION}" --constraint "${CONSTRAINT_URL}"
+```
+5. Install all required dependencies for a project:
+`pip install requirements.txt`
+6. Run standalone command. The Standalone command will initialise the database, make a user, and start all components for you:
```
Comment (drjova) [inline README.md]: Could you please add more details how to do that?

**Comment by drjova** [comment]
@ErnestaP could you please also include https://pre-commit.com/? you can check the configuration https://github.com/cern-sis/cern-academic-training/blob/master/.pre-commit-config.yaml and installation instructions https://github.com/cern-sis/cern-academic-training/#prerequisite

**Comment by drjova** [comment]
@ErnestaP please  squash all the commits and reference the issue in the commit message :)

File: .github/workflows/pytest.yml (hunk 1 above)
Comment (ErnestaP) [reply .github/workflows/pytest.yml]: Because I'm using official airflow image: https://github.com/ErnestaP/workflows-2/blob/workflow-376/Dockerfile#L1

**Comment by drjova** [comment]
could we include code coverage as well? https://about.codecov.io/blog/python-code-coverage-using-github-actions-and-codecov/

File: Dockerfile [hunk 7]
```diff
@@ -0,0 +1,14 @@
+FROM apache/airflow:latest-python3.7
```
Comment (drjova) [inline Dockerfile]: I would also pin the image version
https://hub.docker.com/layers/apache/airflow/2.2.4-python3.7/images/sha256-72a2cdcdabbc622c30940f1a9f262d047fdbbe96d5d7a4e324b8c7ec5ef56171?context=explore

File: .github/workflows/pytest.yml [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+      - uses: actions/checkout@v2
+      - name: Install Python 3
+        uses: actions/setup-python@v1
+        with:
+          python-version: 3.7.12
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r requirements.txt &&
+          pip install -r requirements_airflow.txt
```
Comment (drjova) [inline .github/workflows/pytest.yml]:
```suggestion
          python -m pip install --upgrade pip
          pip install -r requirements_airflow.txt -r requirements.txt

File: requirements.txt [hunk 9]
```diff
@@ -0,0 +1,3 @@
...
+coverage==6.3.2
+pre-commit==2.17.0
```
Comment (drjova) [inline requirements.txt:1]: I would add them in requirements-test.txt (are related to tests right?) except pre-commit

File: .github/workflows/pytest.yml [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+      - name: Run tests with pytest
+        run: pytest
```
Comment (drjova) [inline .github/workflows/pytest.yml]: It's missing the coverage step

**Review by drjova — COMMENTED** [review]
Please squash the commits, and since we're using python 3 we don't need the ``__init__.py`` files, we can delete them.

File: tests/test_dag/test_task2.py [hunk 11]
```diff
@@ -0,0 +1,6 @@
+from dags.test_dag import task_2
+
+
+def test_task2():
+    value = task_2()
+    assert value == "second_task"
```
Comment (drjova) [inline tests/test_dag/test_task2.py:6]: Can we somehow test the DAG as well? https://airflow.apache.org/docs/apache-airflow/stable/best-practices.html#unit-tests

**Review by drjova — APPROVED** [review]

File: setup.cfg [hunk 12]
```diff
@@ -0,0 +1,6 @@
+[flake8]
+# E501 black takes care of the line size
+ignore = D401,W503,E501,E265,E203
+max-line-length = 88
+include = scoap3/
```
Comment (drjova) [inline setup.cfg]: we don't need that for the moment

**Review by drjova — APPROVED** [review]
Just the comment and LGTM

**Pull request merged**